src/Maestro/Maestro.Web/MaestroBarClient.cs

```diff
@@ -0,0 +1,328 @@
+using Microsoft.DotNet.Kusto;
+using Microsoft.DotNet.Maestro.Client.Models;
+using Microsoft.DotNet.Services.Utility;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.TeamFoundation.Build.WebApi.Events;
+using System;
+using System.Data;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Maestro.Web
+{
+    /// <summary>
+    ///     A bar client interface for use by DarcLib which talks directly
+    ///     to the database for diamond dependency resolution.  Only a few features are required.
+    /// </summary>
+    internal class MaestroBarClient : IBarClient
+    {
+        private readonly BuildAssetRegistryContext _context;
+        private readonly KustoClientProvider _kustoClientProvider;
+
+        public MaestroBarClient(BuildAssetRegistryContext context)
+        {
+            _context = context;
+        }
+
+        public MaestroBarClient(BuildAssetRegistryContext context,
+                                IKustoClientProvider kustoClientProvider)
+        {
+            _context = context;
+            _kustoClientProvider = (KustoClientProvider) kustoClientProvider;
+        }
+
+        #region Unneeded APIs
+
+        public Task AddDefaultChannelAsync(string repository, string branch, string channel)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<Channel> CreateChannelAsync(string name, string classification)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<Subscription> CreateSubscriptionAsync(string channelName, string sourceRepo, string targetRepo, string targetBranch,
+            string updateFrequency, bool batchable, List<MergePolicy> mergePolicies)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<Channel> DeleteChannelAsync(int id)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task DeleteDefaultChannelAsync(int id)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task UpdateDefaultChannelAsync(int id, string repository = null, string branch = null, string channel = null, bool? enabled = null)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<Subscription> DeleteSubscriptionAsync(Guid subscriptionId)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<Channel> GetChannelAsync(string channel)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task<IEnumerable<Channel>> GetChannelsAsync(string classification = null)
+        {
+            throw new NotImplementedException();
+        }
+
+        public async Task<IEnumerable<DefaultChannel>> GetDefaultChannelsAsync(string repository = null, string branch = null, string channel = null)
```

Comment:
>GetDefaultChannelsAsync [](start = 55, length = 23)

nit: maybe move methods with implementation to the top so all the not implemented methods are in the bottom

Reply:
I don't disagree with you, but I just want to explain why I did it how I did. So, I started with SubscriptionActorService\MaestroBarClient.cs, where most of the functions aren't implemented. I removed the implementations of the functions I wasn't interested in and wrote the implementations for the ones I was interested in in place. Essentially, this file represents the various swagger api functions as they were added to Maestro, hence the current ordering.